### PR TITLE
expression: Add json_unquote push down for tiflash

### LINKED ISSUE #48986: Add json_unquote push down support for tiflash
## Enhancement
https://github.com/pingcap/tiflash/issues/8334

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #48986 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/expression/expr_to_pb_test.go [hunk 1]
```diff
@@ -540,6 +540,11 @@ func TestExprPushDownToFlash(t *testing.T) {
 	require.NoError(t, err)
 	exprs = append(exprs, function)
 
+	// json_unquote's argument is not cast(json as string)
```
Comment (SeaRise) [inline pkg/expression/expr_to_pb_test.go:543]: how about removing L536 ~ L543?

**Review by SeaRise — APPROVED** [review]
Others LGTM

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**